### ISSUE TITLE
`link-neighbor?`'s behavior conflicts with its dictionary description

### ISSUE THREAD

**Issue by TheBizzle** [NetLogo/NetLogo#917]
The NetLogo dictionary says this about [`link-neighbor?`](http://ccl.northwestern.edu/netlogo/docs/dictionary.html#link-neighbor):

> Reports true if there is an undirected link between _turtle_ and the caller.

however, this language test fails in Headless, NetLogo 5.3, and NetLogo Web:

```
LinkNeighborIsUndirectedOnly3
  directed-link-breed [directors director]
  O> crt 1
  O> crt 1 [ create-directors-from other turtles ]
  [link-neighbor? turtle 0] of turtle 1 => false # actual: true
  [link-neighbor? turtle 1] of turtle 0 => false # actual: true
```

I guess it's not technically a contradiction, since the dictionary only says "if" and not "only if" (not specifying what happens if you don't have an undirected link), but it seems pretty heavily implied that directed links are not supposed to be treated as link neighbors.

I mean, there's not even a `director-neighbor?` primitive generated for the `director` breed in the language test that I provided.  The idea is that you're not supposed to be able to use a directed breed with `<breed>-neighbor?`.  Instead, you're supposed to use the `(in|out)-(link|<breedName>)-neighbor?` prims for checking directed neighborship.

In summary: I think the `link-neighbor?` primitive is giving wrong results in cases where there are only directed neighbors.

**Comment by TheBizzle** [comment]
@nicolaspayette pointed out that this is a duplicate of #739.  Closing.
